elyra/util/operation_parser.py

```diff
@@ -111,53 +99,89 @@ def __init__(self, operation_filepath):
     def type(self):
         raise NotImplementedError()
```

Comment:
This should include `@abstractmethod`.  Also, you don't need to raise `NotImplementedError` if you're properly derived from `ABC`, a simple `pass` is sufficient.  The `ABC` will raise an appropriate exception when an attempt is made to create a subclass that doesn't implement the appropriate methods.